app/util/analytics.py

```diff
@@ -37,35 +31,9 @@ def initialize(eventlog_file=None):
     :param eventlog_file: The filename to output events to, 'STDOUT' to output to stdout, None to disable event logging
     :type eventlog_file: str | None
     """
-    global _analytics_logger, _eventlog_file
-
-    _eventlog_file = eventlog_file
-    if not eventlog_file:
-        _analytics_logger = None
-        return
-
-    if eventlog_file.upper() == 'STDOUT':
-        event_handler = StreamHandler(sys.stdout)
-    else:
-        fs.create_dir(os.path.dirname(eventlog_file))
-        previous_log_file_exists = os.path.exists(eventlog_file)
-
-        event_handler = RotatingFileHandler(
-            filename=eventlog_file,
-            max_size=Configuration['max_eventlog_file_size'],
-            backup_count=Configuration['max_eventlog_file_backups'],
-        )
-        if previous_log_file_exists:
-            event_handler.perform_rollover()  # force starting a new eventlog file on application startup
+    global _event_log
 
-    event_handler.format_string = '{record.message}'  # only output raw log message -- no timestamp or log level
-    handler = TaggingHandler(
-        {'event': event_handler},  # enable logging to the event_handler with the event() method
-        bubble=True,
-    )
-    handler.push_application()
-
-    _analytics_logger = TaggingLogger('analytics', ['event'])
+    _event_log = EventLog(filename=eventlog_file, disable_logging=eventlog_file is None)
```

Comment:
Should the EventLog class just decide for itself if logging is disabled since it already has the value for eventlog_file?